src/app/questionnaire-item/questionnaire-item-generic/questionnaire-item-generic.component.html

```diff
@@ -75,34 +75,68 @@ <h2>
-      </mat-label>
-      <mat-select [formControl]="formControl" [required]="isRequired">
-        <ng-container *ngIf="item.answerOption">
-          <mat-option
-            *ngFor="let option of item.answerOption"
-            [value]="getAnswerOptionValue(option)"
-          >
-            {{ getAnswerOptionDisplay(option) }}
-            <!-- viewValue -->
-          </mat-option>
-        </ng-container>
-        <ng-container *ngIf="item.answerValueSet">
-          <mat-option
-            *ngFor="let concept of getAnswerValueSet()"
-            [value]="getAnswerValueSetValue(concept)"
+    <ng-container *ngSwitchCase="'choice'">
+      <ng-container [ngSwitch]="getItemControlCode()">
+        <ng-container *ngSwitchCase="'radio-button'">
+          <label class="mat-body-2">
+            <app-item-label [item]="item"></app-item-label>
+            <span *ngIf="isRequired">&nbsp;*</span>
+          </label>
+          <mat-radio-group
+            class="choice-radio-button-group"
+            [ngClass]="{ horizontal: getChoiceOrientation() === 'horizontal' }"
+            [formControl]="formControl"
+            [required]="isRequired"
           >
-            {{ getAnswerValueSetDisplay(concept) }}
-          </mat-option>
+            <ng-container *ngIf="item.answerOption">
+              <mat-radio-button
+                *ngFor="let option of item.answerOption"
+                [value]="getAnswerOptionValue(option)"
+              >
+                {{ getAnswerOptionDisplay(option) }}
+                <!-- viewValue -->
+              </mat-radio-button>
+            </ng-container>
+            <ng-container *ngIf="item.answerValueSet">
+              <mat-radio-button
+                *ngFor="let concept of getAnswerValueSet()"
+                [value]="getAnswerValueSetValue(concept)"
+              >
+                {{ getAnswerValueSetDisplay(concept) }}
+              </mat-radio-button>
+            </ng-container>
+          </mat-radio-group>
         </ng-container>
-      </mat-select>
-      <mat-error
-        *ngIf="formControl?.errors?.required"
-        translate="errors.required"
-      ></mat-error>
-    </mat-form-field>
+        <mat-form-field *ngSwitchDefault appearance="outline">
+          <mat-label>
+            <app-item-label [item]="item"></app-item-label>
+          </mat-label>
+          <mat-select [formControl]="formControl" [required]="isRequired">
+            <ng-container *ngIf="item.answerOption">
+              <mat-option
+                *ngFor="let option of item.answerOption"
+                [value]="getAnswerOptionValue(option)"
+              >
+                {{ getAnswerOptionDisplay(option) }}
+                <!-- viewValue -->
+              </mat-option>
+            </ng-container>
+            <ng-container *ngIf="item.answerValueSet">
+              <mat-option
+                *ngFor="let concept of getAnswerValueSet()"
```

Comment:
What does concept mean in this context? I would expect the variable to be named anserValueSet or something similar.

Reply:
I agree that this naming is not as well as it could be. It is due to the naming of the original code (see the mat-select below). The returned values of getAnswerValueSet are in fact concepts, so getAnswerValueSet could be misleading too.
Because nearly the same code appears in multiple locations I will leave for now as it is. This should be overthought in the next big refactoring (e. g. while extracting the form components in to ng-fhirjs or a separate library).